src/Rules/Operators/OperandsInComparisonRule.php

```diff
@@ -0,0 +1,61 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Operators;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\BinaryOp;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Type\FloatType;
+
+class OperandsInComparisonRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return BinaryOp::class;
+	}
+
+	/**
+	 * @param Node $node
+	 * @param Scope $scope
+	 * @return string[]
+	 */
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$node instanceof BinaryOp\Equal
+			&& !$node instanceof BinaryOp\Identical
+			&& !$node instanceof BinaryOp\GreaterOrEqual
+			&& !$node instanceof BinaryOp\SmallerOrEqual
+		) {
+			return [];
+		}
+
+		$rightType = $scope->getType($node->right);
+		$leftType = $scope->getType($node->left);
+
+		if ($rightType instanceof FloatType || $leftType instanceof FloatType) {
+			if ($node instanceof  BinaryOp\Equal || $node instanceof BinaryOp\Identical) {
+				return [
+					'You can not use an exact comparison for floating-point numbers.
+					Should use `abs($left - $right) < $epsilon`, where $epsilon maximum permissible error.',
```

Comment:
1I'd suggest rewording as such:
> Exact comparison of floating-point numbers is not accurate, you should use `abs($a - $b)` < $epsilon where $epsilon is the maximum allowed deviation.

Equivalently for the other two messages.

Reply:
I changed it